src/main/java/org/antipathy/mvn_scalafmt/FormatMojo.java

```diff
@@ -60,4 +54,18 @@ public void execute() throws MojoExecutionException {
             getLog().info("Skip flag set, skipping formatting");
         }
     }
+
+    private List<Object> prepareSources(boolean skip, List<File> sources) throws MojoExecutionException {
+        ArrayList<Object> prepared = new ArrayList<>();
+        if(!skip) {
+            for (File source : sources) {
+                try {
+                    prepared.add(source.getCanonicalFile());
+                } catch (IOException e) {
+                    throw new MojoExecutionException("Can't get canonical file for " + source.toString(), e);
```

Comment:
Could we get the same result by changing
```scala
paths.map(_.toString).filter(p => Files.exists(Paths.get(p)))
```
to 

```scala
paths.map(_.asInstanceOf[File].getCanonicalPath).filter(p => Files.exists(Paths.get(p)))
```
in `getSourcePaths` on `ScalaFormatter`?

Reply:
Alternatively we could change the method to: 

```scala
private[mvn_scalafmt] def getSourcePaths(paths: Seq[Any], log: Log): Seq[String] =
    if (paths == null) {
      Seq[String]()
    } else {
      paths.map(_.asInstanceOf[File].getCanonicalPath).flatMap{ p =>
        if (Files.exists(Paths.get(p))) {
          Some(p)
        } else {
          log.error(s"Could not locate Scala source at $p")
          None
        }
      }
    }
```

Reply:
I agree about getting canonical path inside `getSourcePaths`. It really feels better.

BTW, is there any reason to pass `List[Object]` instead of `List[File]` into `ScalaFormatter.format()`? I think, it's better to avoid this `asInstanceOf` and just pass files as is.

Reply:
Now you mention it, no.  feel free to change that.